app/src/app/store/mapStore.ts

```diff
@@ -95,231 +101,287 @@ export interface MapStore {
   updateVisibleLayerIds: (layerIds: LayerVisibility[]) => void;
   contextMenu: ContextMenuState | null;
   setContextMenu: (menu: ContextMenuState | null) => void;
+  userMaps: Array<DocumentObject & { name?: string }>;
+  setUserMaps: (userMaps: MapStore["userMaps"]) => void;
+  upcertUserMap: (props: {
+    documentId?: string;
+    mapDocument?: MapStore["mapDocument"];
+    userMapDocumentId?: string;
+    userMapData?: MapStore["userMaps"][number];
+  }) => void;
 }
 
-const initialLoadingState = typeof window !== 'undefined' && new URLSearchParams(window.location.search).has(
-  "document_id"
-)
-  ? "loading"
-  : "initializing";
+const initialLoadingState =
+  typeof window !== "undefined" &&
+  new URLSearchParams(window.location.search).has("document_id")
+    ? "loading"
+    : "initializing";
 
-export const useMapStore = create(devtools(
-  subscribeWithSelector<MapStore>((set, get) => ({
-    appLoadingState: initialLoadingState,
-    setAppLoadingState: (appLoadingState) => set({ appLoadingState }),
-    mapRenderingState: "initializing",
-    setMapRenderingState: (mapRenderingState) => set({ mapRenderingState }),
-    getMapRef: () => null,
-    setMapRef: (mapRef) => {
-      set({
-        getMapRef: () => mapRef.current,
-        appLoadingState:
-        initialLoadingState === "initializing"
-            ? "loaded"
-            : get().appLoadingState,
-      })
-    },
-    mapLock: false,
-    setMapLock: (mapLock) => set({ mapLock }),
-    mapDocument: null,
-    setMapDocument: (mapDocument) => {
-      const currentMapDocument = get().mapDocument
-      if (currentMapDocument?.document_id === mapDocument.document_id) {
-        return
-      }
-      get().setFreshMap(true);
-      get().resetZoneAssignments();
-
-      set({
-        mapDocument: mapDocument,
-        shatterIds: { parents: new Set(), children: new Set() },
-      });
-    },
-    shatterIds: {
-      parents: new Set(),
-      children: new Set(),
-    },
-    handleShatter: async (document_id, geoids) => {
-      set({ mapLock: true });
-      const shatterResult = await patchShatter.mutate({
-        document_id,
-        geoids,
-      });
+export const useMapStore = create(
+  persist(
```

Comment:
H: I really don't think we should persist the _entire_ store, no? I mean we could but I was imagining we just persist data that is unique to the user, e.g. the list of user maps.

Reply:
That's super cool- it may be worth us adding a note to the docs to remind where / how that's being set